cmass/summaries/Pk_nbkit.py

```diff
@@ -103,6 +103,22 @@ def main(cfg: DictConfig) -> None:
     logging.info(f'Saving P(k) to {outpath}...')
     np.savez(outpath, k_gal=k_gal, p0k_gal=p0k_gal,
              p2k_gal=p2k_gal, p4k_gal=p4k_gal)
+    
+    # check if use filter
+    if hasattr(cfg, 'filter'):
```

Comment:
This should check if the `..._weight.npy` files exist and load them if they do.

If they don't exist, it should assume the rdz's are unweighted, and pass `None` to `compute_Pk`, which will then intrinsically assign weights of 1 to all galaxies.

Then, you can save it into the same `Pk` folder as previously, but now with the `filter_name` tag in the filename.

This will clean up the function, so you don't need to call `compute_Pk` or the saving logic more than once